pyscript.core/src/plugins/deprecations-manager.js

```diff
@@ -0,0 +1,22 @@
+// PyScript Derepcations Plugin
+import { hooks } from "../core.js";
+import { notify } from "./error.js";
+
+// eslint-disable-next-line no-unused-vars
+hooks.main.onReady.add(function checkDeprecations(pyScript) {
```

Comment:
there are a few issues with this approach:

  * a worker won't trigger this hook ever ... this is only for py-script on the main thread. A solution is to add the same hook, still on main, but as `onWorker` so that whenever a worker script bootstraps the main can notify, if needed
  * scripts added at runtime or lazily due "*reasons*" could happen after another py-script landed ... and being that a different project/source of code I am not sure we need a Mutation Observer
  * we could notify ASAP instead of too late ... so I would actually move this function in the top level, add it to both `hooks.main.onReady` and `hooks.main.onWroker` (note both on `main`) but also execute it ASAP instead of waiting, as scripts with that source are likely already in the page.

thoughts?

Reply:
In short, this is how I would rewrite this instead:

```js
// PyScript Derepcations Plugin
import { hooks } from "../core.js";
import { notify } from "./error.js";

// react lazily on PyScript bootstrap
hooks.main.onReady.add(checkDeprecations);
hooks.main.onWorker.add(checkDeprecations);

// check ASAP once the page is ready
checkDeprecations();

/**
 * Verify there are no scripts from pyscript.net/latest
 */
function checkDeprecations() {
    for (const script of document.querySelectorAll("script"))
        checkLoadingScriptsFromLatest(script.src);
}

/**
 * Check if src being loaded from pyscript.net/latest and display a notification if true
 * @param {string} src
 */
function checkLoadingScriptsFromLatest(src) {
    if (/\/pyscript\.net\/latest/.test(src)) {
        notify(
            "Loading scripts from latest is deprecated and will be removed soon. Please use a specific version instead.",
        );
    }
}
```

I hope this makes sense.

Reply:
aaaaah, yes! Honestly that's kind of what I mentioned in the PR comment. I was rushing (very limited time) and wasn't sure. +1 on your proposed... 

@WebReflection  Only think I'm not sure about the above is around the eager `checkDeprecations()`. It seems a bit redundant and I'm not sure what use cases it's covering that the other `onReady` and `onWorker` calls are not going to catch. I don't think it's a big deal and no harm on keeping it there but did you have something specific in mind?

Reply:
it's just to early show the warning but indeed you can avoid that.